app/commands/mentor/discussion/finish_by_student.rb

```diff
@@ -91,7 +91,8 @@ def award_badge!
       end
 
       def log_metric!
-        Metric::Queue.(:finish_mentoring, discussion.finished_at, discussion:, track:, user: student)
+        Metric::Queue.(:finish_mentoring, discussion.finished_at,
+          discussion:, track:, user: student, remote_ip: Exercism.request_context[:remote_ip])
```

Comment:
Any reason not to move this into Metric::Queue itself?

Reply:
Yes, as there are a couple of metrics (the Github-related ones) that should have `remote_ip` set to `nil`. If I were to remove it here, the logic to determine whether or not has to move elsewhere. There are a couple of options:

1. Have `Metric::Queue` determine whether or not the `remote_ip` should be added. This feels like a violation of the separation of concerns, so I don't really like this.
2. Another would be to add a `remote_ip` method to the `Metric` base class that subclasses can override (which the GitHub ones would do), and then use that when creating the metric. 

I'll try the second approach.

Reply:
Check https://github.com/exercism/website/pull/2769/commits/9e058ef0a5cf6babd6fcdf6e9cdad53861f78b57